llvm/lib/SYCLLowerIR/ComputeModuleRuntimeInfo.cpp

```diff
@@ -152,6 +150,10 @@ PropSetRegTy computeModuleProperties(const Module &M,
     PropSet.add(PropSetRegTy::SYCL_DEVICE_REQUIREMENTS,
                 computeDeviceRequirements(M, EntryPoints).asMap());
   }
+  auto *SpecConstsMD =
+      M.getNamedMetadata(SpecConstantsPass::SPEC_CONST_MD_STRING);
+  bool SpecConstsMet =
+      SpecConstsMD != nullptr && SpecConstsMD->getNumOperands() != 0;
```

Comment:
Both `collectSpecConstantMetadata` and `collectSpecConstantDefaultValuesMetadata` perform this metadata check:

https://github.com/intel/llvm/blob/97231579fdad2c906439162accc0145e20c23e5b/llvm/lib/SYCLLowerIR/SpecConstants.cpp#L1035-L1039

Therefore, I wonder if we just need to omit it all together and instead put `PropSet.add` under condition that `TmpSpecIDMap` is not empty.

Reply:
thanks, ill try this and make a follow-up PR if it works